biz.aQute.bndlib.tests/src/test/baseline/BaselineTest.java

```diff
@@ -555,4 +557,51 @@ public void testMovePackageToBundleClassPath() throws Exception {
 		assertEquals("com.google.ical.iter", change.packageName);
 		assertEquals("20110304.0.0", change.suggestedVersion.toString());
 	}
+
+	// This tests the scenario where a super type is injected into the class
+	// hierarchy but the super class comes from outside the bundle so that the
+	// baseline cannot find it. Since the class hierarchy was cut off, the
+	// baseline would _forget_ that every class inherits from Object, and _lose_
+	// Object's methods if not directly implemented.
+	public void testCutOffInheritance() throws Exception {
+		Processor processor = new Processor();
+
+		DiffPluginImpl differ = new DiffPluginImpl();
+		Baseline baseline = new Baseline(processor, differ);
+
+		Jar older = new Jar(IO.getFile("jar/baseline/inheritance-change-1.0.0.jar"));
+		Jar newer = new Jar(IO.getFile("jar/baseline/inheritance-change-1.1.0.jar"));
```

Comment:
You need to put the Jar constuctors in a try-with-resource to properly close the files.

`try (Jar older = ...; Jar newer = ...) {`